addons/mail/static/src/components/composer_suggested_partners/composer_suggested_partners.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-name="mail.ComposerSuggestedPartners" owl="1">
+        <!-- List of followers -->
+        <div class="o_composer_suggested_partners">
+            <t t-if="state.more" t-foreach="thread.suggestedPartners" t-as="recipient" t-key="partner_id">
+                <div t-attf-title="Add as recipient and follower (reason: {{recipient.reason}})">
+                    <div class="custom-control custom-checkbox">
+                        <input t-on-change="_onChangeSuggestedPartners(recipient_index)" type="checkbox" class="custom-control-input" t-attf-id="o_composer_suggested_partners_p{{recipient_index}}" t-att-checked="recipient.checked ? 'checked' : undefined" t-att-data-fullname="recipient.full_name"/>
+                        <label t-attf-for="o_composer_suggested_partners_p{{recipient_index}}" class="custom-control-label">
```

Comment:
Relying on `recipient_index` here is a sign that this code (at least, I'd say whole item in list) should be in a component by itself.

Reply:
Why ? How is `recipient_index` an issue ?

Reply:
Template must be as declarative as possible. `recipient_index` is a magic variable whose value is based on index in the for-loop, so it's imperative mindset and makes hard to understand. This also makes changes more complicated, and this encourages use of dynamically created strings / handler calls with magic parameter, which are hard to debug. If you isolate in a component, you don't need dynamically evaluated strings/ handler calls with magic parameter, which makes everything better to maintain.